transport/src/main/java/io/netty5/channel/ChannelOutboundBuffer.java

```diff
@@ -362,6 +364,27 @@ void forEachFlushedMessage(Predicate<Object> processor) {
         } while (isFlushedEntry(entry));
     }
 
+    void consumeEachFlushedMessage(BiPredicate<Object, Promise<Void>> processor) {
+        assert executor.inEventLoop();
+
+        requireNonNull(processor, "processor");
+
+        Entry entry = flushedEntry;
+        if (entry == null) {
+            return;
+        }
+
+        boolean keepGoing = true;
+        do {
+            if (!entry.cancelled) {
+                keepGoing = processor.test(entry.msg, entry.promise);
+                decrementPendingOutboundBytes(entry.pendingSize);
```

Comment:
The "downside" of this is that it basically decrement the pending bytes before it actually really "write" them to the transport. Not a big deal but we need to remember to account for it somewhere else then imho.

Reply:
@normanmaurer I moved the decrement to a future listener.

Reply:
ok... my main concern was that we might mark the channel as writable again while we still have the data buffered.

Reply:
I think I can prevent that with `isWriteFlushedScheduled`, but also, let's do the delayed decrement for the time being to be on the safe side. We can look at optimization opportunities once the two code bases are together.